derive_client/clients/base_client.py

```diff
@@ -781,3 +787,280 @@ def transfer_from_subaccount_to_funding(self, amount: int, asset_name: str, suba
             condition=_is_final_tx,
             transaction_id=withdraw_result.transaction_id,
         )
+
+    def _extract_transaction_id(self, response_data: dict) -> str:
+        """
+        Extract transaction ID from response data.
+
+        Args:
+            response_data (dict): The response data from an API call
+
+        Returns:
+            str: The transaction ID
+
+        Raises:
+            ValueError: If no valid transaction ID is found in the response
+        """
+        if "result" in response_data and "transaction_id" in response_data["result"]:
+            transaction_id = response_data["result"]["transaction_id"]
+            if transaction_id:
+                return transaction_id
+        raise ValueError("No valid transaction ID found in response")
+
+    def transfer_position(
+        self,
+        instrument_name: str,
+        amount: float,
+        limit_price: float,
+        from_subaccount_id: int,
+        to_subaccount_id: int,
+        position_amount: float,
+    ) -> DeriveTxResult:
+        """
+        Transfer a single position between subaccounts.
+
+        Parameters:
+            instrument_name (str): The name of the instrument to transfer.
+            amount (float): The amount to transfer (absolute value). Must be positive.
+            limit_price (float): The limit price for the transfer. Must be positive.
+            from_subaccount_id (int): The subaccount ID to transfer from.
+            to_subaccount_id (int): The subaccount ID to transfer to.
+            position_amount (float): The original position amount to determine direction.
+                                   Must be provided explicitly (use get_positions() to fetch current amounts).
+
+        Returns:
+            DeriveTxResult: The result of the transfer transaction.
+
+        Raises:
+            ValueError: If amount, limit_price are not positive, position_amount is zero, or if instrument not found.
+        """
+        # Validate inputs
+        if amount <= 0:
```

Comment:
Isnt this already going to raise when the amoutns and prices are put into the position object?